test/integration_test/px_statfs_webhook_test.go

```diff
@@ -0,0 +1,154 @@
+		// Filesystem                     Type 1K-blocks  Used Available Use% Mounted on
+		// /dev/pxd/pxd585819943023766088 ext4  51343840 54556  48651460   1% /sv4test
+		//
+		//
+		// NFS mount sample output:
+		//
+		// $ kubetl exec -it virt-launcher-sim-dep-8476fffd8b-f7vbk -c sv4test -- df -T /sv4test
+		// Filesystem                                          Type 1K-blocks  Used Available Use% Mounted on
+		// 192.168.121.49:/var/lib/osd/pxns/585819943023766088 nfs   51344384 54272  48652288   1% /sv4test
+
+		output := runCommandInSv4TestContainer(t, &pod, []string{"df", "-T", sharedMountPath})
+		isBindMount := regexp.MustCompile(`pxd.*ext4`).MatchString(output)
+		if !isBindMount {
+			// must be an NFS mount
+			require.True(t, regexp.MustCompile(`pxns.*nfs`).MatchString(output))
+		}
+		logrus.Infof("isBindMount=%v", isBindMount)
+
+		// check statfs() on the PX volume
+		output = runCommandInSv4TestContainer(t, &pod, []string{"stat", "--format=%T", "-f", sharedMountPath})
+		output = strings.TrimSpace(output)
+		logrus.Infof("statfs(%v)=%v", sharedMountPath, output)
+
+		if isVirtLauncherContext(ctx) {
+			// should always be "nfs"
+			require.Equal(t, output, "nfs",
+				"statfs() did not return 'nfs' for virt launcher pod %s/%s", pod.Namespace, pod.Name)
+		} else if isBindMount {
+			require.NotEqual(t, output, "nfs",
+				"statfs() returned 'nfs' for bind-mounted pod %s/%s", pod.Namespace, pod.Name)
+		} else {
+			require.Equal(t, output, "nfs",
+				"statfs() did not return 'nfs' for nfs-mounted pod %s/%s", pod.Namespace, pod.Name)
+		}
+
+		// Path /local-vol is an {emptydir} volume so it should never return nfs for the file system type
+		output = runCommandInSv4TestContainer(t, &pod, []string{"stat", "--format=%T", "-f", localMountPath})
+		output = strings.TrimSpace(output)
+		logrus.Infof("statfs(%v)=%v", localMountPath, output)
+		require.NotEqual(t, output, "nfs",
+			"statfs() returned 'nfs' for local volume in pod %s/%s", pod.Namespace, pod.Name)
+
+		if isBindMount {
+			foundBindMount = true
+		} else {
+			foundNFSMount = true
+		}
+	}
+
+	// Sanity check: since we run 1 pod on each of the worker nodes, we expect to find
+	// 1 bind-mounted pod and at least 1 nfs-mounted pod.
+	require.True(t, foundBindMount, "bind-mounted pod not found for context %s", ctx.App.Key)
+	require.True(t, foundNFSMount, "nfs-mounted pod not found for context %s", ctx.App.Key)
+	logrus.Infof("validated statfs for context %v", ctx.App.Key)
+}
+
+func runCommandInSv4TestContainer(t *testing.T, pod *corev1.Pod, cmd []string) string {
+	container := "sv4test"
+	output, err := core.Instance().RunCommandInPod(cmd, pod.Name, container, pod.Namespace)
+	require.NoError(t, err,
+		"failed to run command %v inside the pod %v/%v", cmd, pod.Namespace, pod.Name)
+	return output
+}
+
+func isVirtLauncherContext(ctx *scheduler.Context) bool {
+	return strings.HasPrefix(ctx.App.Key, "virt-launcher-sim")
```

Comment:
Any reason not to just check `kubevirt.io: virt-launcher`?

Reply:
That should also work. I originally had these tests in torpedo repo where I had to check for this condition on a context just to decide whether to schedule it or not (i.e. before any pods started running). We don't have that here. But in any case, I slightly prefer to check it this way because it expresses the intent correctly. If the virt-launcher-sim app does not have the label, the test will fail and point out the problem. Otherwise, the test will pass without actually testing anything.